### PR TITLE
`data_factory_linked_service_sql_server_resource` supports userName

### PR DESCRIPTION
**Pull request by heiazuo**
### create a public pull request for `data_factory_linked_service_sql_server_resource` supports userName
Link to https://github.com/hashicorp/terraform-provider-azurerm/issues/15475
**test result：**
image

### CODE REVIEW

File: internal/services/datafactory/data_factory_linked_service_sql_server_resource.go [hunk 1]
```diff
@@ -266,6 +275,7 @@ func resourceDataFactoryLinkedServiceSQLServerRead(d *pluginsdk.ResourceData, me
 	}
 
 	if properties := sqlServer.SQLServerLinkedServiceTypeProperties; properties != nil {
+		d.Set("user_name", properties.UserName)
```
Comment (catriona-m) [inline internal/services/datafactory/data_factory_linked_service_sql_server_resource.go:278]: should this be nil checked before setting?

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks @heiazuo just left one small comment. Also, looks like there could be some linting to resolve here?

File: internal/services/datafactory/data_factory_linked_service_sql_server_resource.go (hunk 1 above)
Comment (heiazuo) [reply internal/services/datafactory/data_factory_linked_service_sql_server_resource.go:278]: Tanks for this comment @catriona-m，`nil` will be checked before setting in the `Set` function

**Review by katbyte — APPROVED** [review]
LGTM 🚜

**Pull request merged**